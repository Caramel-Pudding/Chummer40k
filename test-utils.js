import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore, combineReducers } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import characteristics from "./redux/features/characteristics/slice";
import descriptors from "./redux/features/descriptors/slice";

// Import your own reducer
const reducer = combineReducers({ descriptors, characteristics });

function render(
  ui,
  {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
