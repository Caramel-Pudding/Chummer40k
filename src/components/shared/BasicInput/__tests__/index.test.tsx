import React, { FC, useState } from "react";
import { render, cleanup } from "@/tests/test-utils";

import { BasicInput, BasicInputProps } from "..";

describe("<BasicInput/>", () => {
  afterEach(cleanup);

  const testLabel = "TEST";
  const TestWrapper: FC<Partial<BasicInputProps>> = () => {
    const [inputValue, setInputValue] = useState("");

    return (
      <BasicInput
        handler={setInputValue}
        labelText={testLabel}
        value={inputValue}
      />
    );
  };

  it("Basic snapshot test", () => {
    // * #TEST: ARRANGE
    const { container } = render(<TestWrapper />, {});
    // * #TEST: ASSERT
    expect(container).toMatchSnapshot();
  });

  it("should render proper label", () => {
    // * #TEST: ARRANGE
    const { container, getByText } = render(<TestWrapper />, {});
    const label = getByText(testLabel);
    // * #TEST: ASSERT
    expect(label).toHaveTextContent(testLabel);
    expect(container).toMatchSnapshot();
  });

  it("should render undefined label as ''", () => {
    // * #TEST: ARRANGE
    const strange = render(<TestWrapper labelText={undefined} />, {});
    const common = render(<TestWrapper labelText="" />, {});

    // * #TEST: ASSERT
    expect(strange.container).toEqual(common.container);
    expect(strange.container).toMatchSnapshot();
    expect(common.container).toMatchSnapshot();
  });
});
