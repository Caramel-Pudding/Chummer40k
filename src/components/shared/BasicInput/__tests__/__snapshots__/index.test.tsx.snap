// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicInput/> Basic snapshot test 1`] = `
<div>
  <label
    class="flex flex-row justify-between"
  >
    TEST
    <input
      class=""
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<BasicInput/> should render proper label 1`] = `
<div>
  <label
    class="flex flex-row justify-between"
  >
    TEST
    <input
      class=""
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<BasicInput/> should render undefined label as '' 1`] = `
<div>
  <label
    class="flex flex-row justify-between"
  >
    TEST
    <input
      class=""
      type="text"
      value=""
    />
  </label>
</div>
`;

exports[`<BasicInput/> should render undefined label as '' 2`] = `
<div>
  <label
    class="flex flex-row justify-between"
  >
    TEST
    <input
      class=""
      type="text"
      value=""
    />
  </label>
</div>
`;
